src/compas_viewer/scene/meshobject.py

```diff
@@ -42,42 +51,68 @@ class MeshObject(ViewerSceneObject, BaseMeshObject):
     """
 
     def __init__(
-        self, mesh: Mesh, hide_coplanaredges: Optional[bool] = None, use_vertexcolors: Optional[bool] = None, **kwargs
+        self,
+        mesh: Mesh,
+        vertexcolor: Optional[ColorDictValueType] = None,
+        edgecolor: Optional[ColorDictValueType] = None,
+        facecolor: Optional[ColorDictValueType] = None,
+        hide_coplanaredges: Optional[bool] = None,
+        use_vertexcolors: Optional[bool] = None,
+        **kwargs
     ):
         super().__init__(mesh=mesh, **kwargs)
+
         self.mesh: Mesh
+
         self.hide_coplanaredges = (
-            hide_coplanaredges if hide_coplanaredges is not None else self.config.hide_coplanaredges
+            hide_coplanaredges if hide_coplanaredges is not None else self.viewer.config.hide_coplanaredges
         )
-        self.use_vertexcolors = use_vertexcolors if use_vertexcolors is not None else self.config.use_vertexcolors
-        self.vertexcolor = {
-            vertex: self.mesh.vertex_attribute(vertex, "color")
-            or self.facescolor.get(vertex, self.facescolor["_default"])  # type: ignore
-            for vertex in self.mesh.vertices()
-        }
-
-    def _read_points_data(self) -> DataType:
+        self.use_vertexcolors = (
+            use_vertexcolors if use_vertexcolors is not None else self.viewer.config.use_vertexcolors
+        )
+
+        if not vertexcolor:
+            self.vertexcolor = self.viewer.config.pointscolor
+            for vertex in self.mesh.vertices():
+                self.vertexcolor[vertex] = self.mesh.vertex_attribute(vertex, "color")  # type: ignore
+        else:
+            self.vertexcolor = vertexcolor
+
+        if not edgecolor:
+            self.edgecolor = self.viewer.config.linescolor
+            for u, v in self.mesh.edges():
+                self.edgecolor[(u, v)] = self.mesh.edge_attribute((u, v), "color")  # type: ignore
+        else:
+            self.edgecolor = edgecolor
+
+        if not facecolor:
+            self.facecolor = self.viewer.config.facescolor
+            for face in self.mesh.faces():
+                self.facecolor[face] = self.mesh.face_attribute(face, "color")  # type: ignore
+        else:
+            self.facecolor = facecolor
+
+    def _read_points_data(self) -> ShaderDataType:
         positions = []
         colors = []
         elements = []
         i = 0
 
         for vertex in self.mesh.vertices():
-            assert isinstance(vertex, int)
             positions.append(self.mesh.vertex_coordinates(vertex))
-            colors.append(self.pointscolor.get(vertex, self.pointscolor["_default"]))  # type: ignore
+            colors.append(self.vertexcolor[vertex] or self.vertexcolor.default)  # type: ignore
```

Comment:
`or self.vertexcolor.default` is not necessary. that is the default behaviour...

Reply:
That did not work when i tried. sometimes `vertexcolor[vertex]` does exist but is `None`, thus returns a `None`